File: Submissions/ABC/ABC086/C/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);

    int N = sc.nextInt();
    int[] t = new int[N];
    int[] x = new int[N];
    int[] y = new int[N];

    for (int i = 0; i < N; i++) {
      t[i] = sc.nextInt();
      x[i] = sc.nextInt();
      y[i] = sc.nextInt();
    }

    sc.close();

    String result = "Yes";

    for (int i = 0; i < N; i++) {

      if (t[i] < x[i] + y[i]) {
        result = "No";
        break;
      } else {
        if (((x[i] + y[i]) % 2) != (t[i] % 2)) {
          result = "No";
          break;
        }
      }
    }

    System.out.println(result);

  }

}

File: Submissions/ABC/ABC086/C/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

  int N;
  cin >> N;

  vector<int> t(N);
  vector<int> x(N);
  vector<int> y(N);
  
  for (int i = 0; i < N; i++) {
    cin >> t[i] >> x[i] >> y[i];
  }

  string result = "Yes";

  for (int i = 0; i < N; i++) {
    
    int distance = x[i] + y[i];
    
    if (t[i] < distance) {
      result = "No";
      break;
    } else if (distance % 2 != t[i] % 2) {
      result = "No";
      break;
    }
  }

  cout << result << endl;

}